src/libasr/runtime/lfortran_intrinsics.c

```diff
@@ -1165,3 +1165,80 @@ LFORTRAN_API void _lpython_close(int64_t fd)
         exit(1);
     }
 }
+
+#include <unwind.h>
+#include <link.h>
+
+static _Unwind_Reason_Code unwind_callback(struct _Unwind_Context *context,
+        void *vdata) {
+    struct Stacktrace *d = (struct Stacktrace *) vdata;
+    uintptr_t pc = _Unwind_GetIP(context);
+    if (pc != 0) {
+        pc--;
+        if (d->size < LCOMPILERS_MAX_STACKTRACE_LENGTH) {
+            d->pc[d->size] = pc;
+            d->size++;
+        } else {
+            // TODO: Do not know how to report an error here
+        }
+    }
+    return _URC_NO_REASON;
+}
+
+struct Stacktrace get_stacktrace_addresses()
+{
+    struct Stacktrace d;
+    d.size = 0;
+    _Unwind_Backtrace(unwind_callback, &d);
+    return d;
+}
+
+struct dl_phdr_info {
+    ElfW(Addr) dlpi_addr;
+    const char *dlpi_name;
+    const ElfW(Phdr) *dlpi_phdr;
+    ElfW(Half) dlpi_phnum;
+};
+
+extern int dl_iterate_phdr (int (*__callback) (struct dl_phdr_info *,
+    size_t, void *), void *__data);
+
+int count = 0;
+
+int shared_lib_callback(struct dl_phdr_info *info,
+        size_t /* size */, void *_data) {
+    struct Stacktrace *d = (struct Stacktrace *) _data;
+    for (int i = 0; i < info->dlpi_phnum; i++) {
+        if (info->dlpi_phdr[i].p_type == PT_LOAD) {
+            ElfW(Addr) min_addr = info->dlpi_addr + info->dlpi_phdr[i].p_vaddr;
+            ElfW(Addr) max_addr = min_addr + info->dlpi_phdr[i].p_memsz;
+            printf("%lx\n", d->current_pc);
```

Comment:
This prints `0`, I think the value is not passed correctly.
@certik any idea why does this prints `0`?